src/components/ResourceBrowser/ResourceList/ColumnSelector.tsx

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2024. Devtron Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState, useMemo, useRef } from 'react'
+import ReactSelect, { components, ValueContainerProps, MenuListProps } from 'react-select'
+import { Option } from '@devtron-labs/devtron-fe-common-lib'
+import { ReactComponent as Setting } from '@Icons/ic-nav-gear.svg'
+import { containerImageSelectStyles } from '../../CIPipelineN/ciPipeline.utils'
+import { OPTIONAL_NODE_LIST_HEADERS } from '../Constants'
+import { ColumnFilterContextType, ColumnSelectorType } from '../Types'
+import { saveAppliedColumnsInLocalStorage } from './utils'
+
+const ColumnFilterContext = React.createContext<ColumnFilterContextType>(null)
+
+export function useColumnFilterContext() {
+    const context = React.useContext(ColumnFilterContext)
+
+    if (!context) {
+        throw new Error(`cannot be rendered outside the component`)
+    }
+
+    return context
+}
+
+const ValueContainer = (props: ValueContainerProps) => {
```

Comment:
Since there are alot of changes can we use SelectPicker instead for column selector? @uxarya-d